tests/unit/test_charm.py

```diff
@@ -466,3 +468,154 @@ def test_given_limit_to_one_filter_when_given_two_csr_then_filtered(  # noqa: E5
         ]
         filter = LimitToOneRequest()
         assert filter.evaluate(b"", 1, requirer_csrs) is False
+
+    def test_given_allowlist_config_filter_when_config_set_then_filter_available(  # noqa: E501
+        self,
+    ) -> None:
+        self.harness.update_config(
+            {
+                "allowed-dns": r"myapp-([0-9]+)?\.mycompany\.com",
+                "allowed-ips": r"172\.25\.0\.[0-9]*",
+                "allowed-oids": r"1\.3\.6\.1\.4\.1\.28978\.[0-9.]*",
+                "allowed-common-name": r"myapp-([0-9]+)?\.mycompany\.com",
+                "allowed-organization": r"Canonical Ltd\.",
+                "allowed-email": r".*@canonical\.com",
+                "allowed-country-code": "(UK|CA|PL|AE|HU|FR|TR|IT)$",
+            }
+        )
+        assert len(self.harness.charm._get_csr_filters()) > 0
+        assert isinstance(self.harness.charm._get_csr_filters()[0], AllowedFields)
+
+    def test_given_allowlist_config_filter_when_config_set_then_filter_applied_properly(  # noqa: E501
+        self, caplog: pytest.LogCaptureFixture
+    ) -> None:
+        # TODO: parametrize csr generating fields
```

Comment:
I think this should be done before merging.

Reply:
done